### DISCUSSION TITLE
Train on non-image datasets

### DISCUSSION THREAD

**Discussion by pfeatherstone** [ultralytics/yolov5#8693] [Ideas]
Is there a way to use this repository on non-image datasets? Like can I plug in a custom data loader that returns 2D objects (not representing images though) and bounding boxes and the repo will train correctly? Things like image augmentation would need to be turned off, anything that assumes the content is image-like should be turned off,... Basically does this repo provide a kind of API which can be used in user-code or is it more meant to be used as a command line tool like darknet?

**Comment by glenn-jocher** [comment]
@pfeatherstone not sure I understand the question, but yes you can modify the dataloaders as you see fit for your custom task. The dataloaders are here:
https://github.com/ultralytics/yolov5/blob/master/utils/dataloaders.py

**Reply by pfeatherstone** [reply]
Is there a not-too-hacky way of using this repo on non-image data? I'm happy to fork and start making aggressive changes. But if you said there were command line options or even better an API I could use, I would use that.

**Reply by pfeatherstone** [reply]
Do you have plans of building an API with re-usable components or is this always going to be a command line tool ? Thanks

**Reply by glenn-jocher** [reply]
@pfeatherstone yes we have plans to launch an API later this year
